src/battle_ai_util.c

```diff
@@ -847,6 +848,41 @@ bool32 CanTargetFaintAi(u32 battlerDef, u32 battlerAtk)
     return FALSE;
 }
 
+u32 NoOfHitsForTargetToFaintAI(u32 battlerDef, u32 battlerAtk)
+{
+    u32 i;
+    u32 currNumberOfHits = 0;
+    u32 maxNumberOfHits = 0;
+
+    for (i = 0; i < MAX_MON_MOVES; i++)
+    {
+        currNumberOfHits = GetNoOfHitsToKOBattler(battlerDef, battlerAtk, i);
+        if (currNumberOfHits > maxNumberOfHits)
+             maxNumberOfHits = currNumberOfHits;
```

Comment:
I think this might be a little backwards, shouldn't it take the smallest number of hits?
```suggestion
    u32 maxNumberOfHits = UINT32_MAX;

    for (i = 0; i < MAX_MON_MOVES; i++)
    {
        currNumberOfHits = GetNoOfHitsToKOBattler(battlerDef, battlerAtk, i);
        if (currNumberOfHits < maxNumberOfHits)
             maxNumberOfHits = currNumberOfHits;
```
EDIT: Although I see there's also `0` hits meaning unknown, so I'm not sure what to do with that. Maybe a `#define UNKNOWN_NO_OF_HITS UINT32_MAX`?

Reply:
That's a huge mistake on my part. 

I like the idea of a `#define UNKNOWN_NO_OF_HITS UINT32_MAX` define.

Reply:
Where would be a good place do add the `UNKNOWN_NO_OF_HITS` define?

Reply:
hm, I have trouble getting it to work with `UINT32_MAX`. All my tests fails with it.

Reply:
Pushed a change with `UNKNOWN_NO_OF_HITS` as 0 for no.

Reply:
> hm, I have trouble getting it to work with `UINT32_MAX`. All my tests fail with it.

I wonder if it would work with something like `255` :thinking: Could be that somewhere in the code we treat it as signed, or put it in something that's smaller than 32-bit.

> Where would be a good place do add the `UNKNOWN_NO_OF_HITS` define?

Good question! Maybe one of the `include/battle_ai_*.h` files? Possibly `battle_ai_main.h`?